src/dmd/mars.d

```diff
@@ -1188,7 +1188,14 @@ void addDefaultVersionIdentifiers(const ref Param params)
         VersionCondition.addPredefinedGlobalIdent("Posix");
         VersionCondition.addPredefinedGlobalIdent("linux");
         VersionCondition.addPredefinedGlobalIdent("ELFv1");
-        VersionCondition.addPredefinedGlobalIdent("CRuntime_Glibc");
+        // Note: This should be done with a triplet, to support cross compilation
+        // However DMD currently does not support it, so this is a simple
+        // fix to make DMD compile on Musl-based systems such as Alpine.
+        // See https://github.com/dlang/dmd/pull/8020
+        version (CRuntime_Musl)
+            VersionCondition.addPredefinedGlobalIdent("CRuntime_Musl");
+        else
+            VersionCondition.addPredefinedGlobalIdent("CRuntime_Glibc");
```

Comment:
... even with this.

Reply:
What do you mean ?

Reply:
I mean I get why you had the version in the if above, i.e. because if this.